salt/_logging/impl.py

```
@@ -400,51 +512,562 @@ def makeRecord(
+    # Nothing below this level should be processed by python's logging machinery
+    logging.root.setLevel(log_level)
+
+
+def set_lowest_log_level_by_opts(opts):
+    """
+    Set the lowest log level by the passed config
+    """
+    log_levels = [
+        get_logging_level_from_string(opts.get("log_level")),
+        get_logging_level_from_string(opts.get("log_level_logfile")),
+    ]
+    for log_level in opts.get("log_granular_levels", {}).values():
+        log_levels.append(get_logging_level_from_string(log_level))
+
+    log_level = min(log_levels)
+    set_lowest_log_level(log_level)
+
+
+def __global_logging_exception_handler(
+    exc_type,
+    exc_value,
+    exc_traceback,
+    _logger=logging.getLogger(__name__),
+    _stderr=sys.__stderr__,
+    _format_exception=traceback.format_exception,
+):
+    """
+    This function will log all un-handled python exceptions.
+    """
+    if exc_type.__name__ == "KeyboardInterrupt":
+        # Do not log the exception or display the traceback on Keyboard Interrupt
+        return
+
+    # Log the exception
+    msg = "An un-handled exception was caught by Salt's global exception handler:"
+    try:
+        msg = "{}\n{}: {}\n{}".format(
+            msg,
+            exc_type.__name__,
+            exc_value,
+            "".join(_format_exception(exc_type, exc_value, exc_traceback)).strip(),
+        )
+    except Exception:  # pylint: disable=broad-except
+        msg = "{}\n{}: {}\n(UNABLE TO FORMAT TRACEBACK)".format(
+            msg,
+            exc_type.__name__,
+            exc_value,
+        )
+    try:
+        _logger.error(msg)
+    except Exception:  # pylint: disable=broad-except
+        # Python is shutting down and logging has been set to None already
+        try:
+            _stderr.write(msg + "\n")
+        except Exception:  # pylint: disable=broad-except
+            # We have also lost reference to sys.__stderr__ ?!
+            print(msg)
```

**question** should we also except here? Or just :shrug: and say, "You deal with it?" :joy:

---

If it's that bad, let it blow up.